go-controller/pkg/ovn/gress_policy.go

```diff
@@ -77,12 +84,34 @@ func newGressPolicy(policyType knet.PolicyType, idx int, namespace, name string)
 		policyName:        name,
 		policyType:        policyType,
 		idx:               idx,
-		peerV4AddressSets: sets.String{},
-		peerV6AddressSets: sets.String{},
+		peerV4AddressSets: &sync.Map{},
+		peerV6AddressSets: &sync.Map{},
 		portPolicies:      make([]*portPolicy, 0),
 	}
 }
 
+func syncMapToSortedList(m *sync.Map) []string {
+	result := []string{}
+	m.Range(func(key, _ interface{}) bool {
+		result = append(result, key.(string))
+		return true
+	})
+	sort.Slice(result, func(i, j int) bool {
```

Comment:
To use sort.Sort I need to define this inerface
```
type Interface interface {
	// Len is the number of elements in the collection.
	Len() int

	// Less reports whether the element with index i
	// must sort before the element with index j.
	
	Less(i, j int) bool

	// Swap swaps the elements with indexes i and j.
	Swap(i, j int)
}
```
and  sort.Slice also uses quickSort, but allows to pass less function directly instead of implementing an interface
```
func Slice(x any, less func(i, j int) bool) {
	rv := reflectValueOf(x)
	swap := reflectSwapper(x)
	length := rv.Len()
	quickSort_func(lessSwap{less, swap}, 0, length, maxDepth(length))
}

// Auto-generated variant of sort.go:quickSort
func quickSort_func(data lessSwap, a, b, maxDepth int) {
```

So the only difference between these options is Slice has 2 more calls:
```
	rv := reflectValueOf(x)
	swap := reflectSwapper(x)
```
But I can just define this interface to stick with the previous implementation

Reply:
wow...sorry to make you go through that. I saw func Sort(data Interface) { and thought it was a regular interface, not some custom type. What bad naming. Anyway, would it be easier to just convert your slice to a strings.set and then get List and Len for free?

Reply:
nvm it looks like what you added is not too much code.